### ISSUE TITLE
[BUG] AttachDynamicAreaToVehicle

### ISSUE THREAD

**Issue by IstuntmanI** [samp-incognito/samp-streamer-plugin#120]
I don't remember that I had this problem before, one of my players reported it now: when there's a driver in vehicle, the area isn't there, if there's only a passenger everything is ok, but immediately after the driver gets out, the other player instantly enters the area if he is in range of the offsets (I use a circle).

**Comment by samp-incognito** [comment]
Are you using offsets? It could be related to #68. GetVehicleRotationQuat only works while the vehicle has a driver, so if the area suddenly starts working for you when the driver steps out, then something might be wrong with the rotation code. I don't think it was ever thoroughly tested.

**Comment by IstuntmanI** [comment]
It is related to it for sure. Yes, I use offsets:
```pawn
new liArea = CreateDynamicCircle( 0.0, 0.0, 2.0, -1, -1, playerid );
AttachDynamicAreaToVehicle( liArea, vehicleid, 0.0, -4.20, 0.0 );
```

You should try looking at the part of code when it is using the quaternion instead of Z angle.

https://github.com/samp-incognito/samp-streamer-plugin/blob/master/src/streamer.cpp#L1010-1012
https://github.com/samp-incognito/samp-streamer-plugin/blob/master/src/utility/geometry.cpp#L142-148

**Comment by IstuntmanI** [comment]
I tested this again:
- No driver: area is at the **back** of the vehicle, *like it should*.
- Driver: area is at the **front** of the vehicle, so it is at the *opposite side*.

**Comment by samp-incognito** [comment]
So the offset is correct, just positioned on the wrong (opposite) side?

**Comment by IstuntmanI** [comment]
Sorry for the late response.

I just tested it with lots of offsets (only for circle and sphere, I don't know if this bug appears for other shapes) and this is what I saw:
- X and Z are normal with or without driver.
- Y is normal without a driver, but is inverted with a driver. (a -4.0 offsets behaves like a 4.0 offset)

**Comment by samp-incognito** [comment]
That's very helpful, thanks. I'll fix this soon.

**Comment by IstuntmanI** [comment]
I updated to v2.9.1 locally and it looks like it is still bugged. Can anyone confirm this ?

**Comment by IstuntmanI** [comment]
I debugged the functions Utility::projectPoint for ```int``` (heading) and ```Vector4f``` (quaternion).

Not in vehicle (using heading/```int```):
```
Utility::projectPoint: int: position is -293.765137, -2226.680908, 29.682093
```

In vehicle (using quaternion/```Vector4f```):
```
Utility::projectPoint: Vector4f: position is -298.131439, -2226.681152, 29.633886 | + -2.182913, -0.722889, -0.048206
```

I inverted the X (which was already inverted here, so I removed the inversation 😆 ) and I got the correct values. I can't understand how your commit didn't fix the bug reproduced and explained above by me (unless I didn't test all cases then).

Seeing that no one cares about this issue, I'll send a commit to my pull request and hope that this is really correct. I tested it.
